cpp/src/utilities/graph_utils.cuh

```diff
@@ -517,5 +517,45 @@ namespace cugraph
                          init);
         cudaCheckError();
     }
+    
+    template<typename DistType>
+	    bool has_negative_val(DistType* arr, size_t n){
+		    // custom kernel with boolean bitwise reduce may be 
+		    // faster.
+#if 0
+		    // cub throws errors with double in cuda-memcheck
+		    // switch to thrust until resolved
```

Comment:
Should we fill an issue in CUB repo?

Reply:
I think so. I didn't investigate it too much though. This was reported only by cuda-memcheck. There were no other side-effects.

Reply:
I tend to trust cuda memcheck, some real bug can be silent for a long time. It would be ideal if you could find a quick repro to fill a CUB issue after you are done with the rest of the SSSP project.